bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,291 @@
+
+; Input Counter
+: [Artifact] The length of the transaction input list, aka `txin_count` (see [[https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki|BIP144]]).BIP-144)
+
+; Input Index
+: [Concept] The position of an input in a transaction's Input List
+
+; Input Script
+: [Component] aka “scriptSig”. The Bitcoin code base used Hungarian notation when it was published.  This transaction field appearing in Transaction Inputs was named for containing the signature and being of the type script.  Today, this field is better understood as the _input script_ which is used to satisfy spending conditions for non-segwit outputs, redirects validation to the Witness Stack for wrapped segwit outputs, and is empty for native segwit outputs. The Input Script may contain an actual script when spending a Bare Output Script, but may only contain push operations for any output type since P2SH.
+
+; Leaf Script
+: [Component] A Condition Script used for spending P2TR outputs via the script path. The Leaf Script appears as a Witness Item.
+
+; Leaf Version
+: [Component] The version of Script used in a leaf script. At the time of writing, only _Tapscript_ is defined.
+
+; Lock Time
+: [Component] aka “nLockTime”. Part of the Transaction Header. Lock Time specifies a time (in blockheight or UNIX time) before which a transaction cannot be included in a block if any input's sequence number indicates non-finality.  The lock time can also indicate a relative spending lock in combination with `OP_CHECKSEQUENCEVERIFY`. Serialized as the four-byte <code>nlock time</code> integer field.
+
+; Marker
+: [Artifact] 1-byte serialization artifact indicating that a type of extended serialization is being used for this transaction.  Must always be <code>00</code> (see [[https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki|BIP144]]).BIP144).  (Note: Non-segwit nodes will only accept _stripped segwit transactions_, because the marker appears at the position where non-segwit nodes expect the input counter. The input counter may not be zero, so a complete (non-stripped) segwit transaction appears invalid to a non-segwit node.)
+
+; Outpoint
+: [Component] Identifies the UTXO being spent in a transaction input.  Consists of a txid and Output Index. The txid is serialized in little endian but displayed in big endian.
+
+; Output Counter
+: [Artifact] The length of the transaction output list, aka `txout_count` (see [[https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki|BIP144]]).BIP-144)
+
```

Comment:
Is this list just pertaining to tx components? Or would we list something like output descriptors here? It would be nice to have a sub-section for common wallet terms such as xpubs, BIP32 paths, etc...

Reply:
I was hoping to restrict myself to aspects of transactions